File: 백준/Gold/2015. 수들의 합 4/수들의 합 4.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        int[] inputData = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            inputData[i] = Integer.parseInt(st.nextToken());
        }

        Map<Long, Integer> sumDict = new HashMap<>();
        sumDict.put(0L, 1);

        long sumNum = 0;
        long answer = 0;

        for (int num : inputData) {
            sumNum += num;

            if (sumDict.containsKey(sumNum - k)) {
                answer += sumDict.get(sumNum - k);
            }

            sumDict.put(sumNum, sumDict.getOrDefault(sumNum, 0) + 1);
        }

        System.out.println(answer);
    }
}

File: 백준/Gold/2015. 수들의 합 4/수들의 합 4.py
import sys
input = sys.stdin.readline


n, k = map(int, input().split())
input_data = list(map(int, input().split()))

sum_dict = {0: 1}

sum_num = 0
answer = 0

for i in input_data:
    sum_num += i
    
    if sum_num - k in sum_dict.keys():
        answer += sum_dict[sum_num - k]

    sum_dict[sum_num] = sum_dict.get(sum_num, 0) + 1

print(answer)